### DISCUSSION TITLE
How to render an inline diagram with back arrows?

### DISCUSSION THREAD

**Discussion by danielvartan** [mermaid-js/mermaid#5886] [Q&A]
Hi everyone,

I'm trying to reproduce the following diagram in Mermaid, but I haven't had any success so far. Could anyone help me out with this?

### The image

![image](https://github.com/user-attachments/assets/2c9d281e-a317-4f32-a164-8f628be40275)

### What I've tried

```{mermaid}
flowchart LR
  A(Real World) -- Design --> B(Conceptual Model)
  B -- Construction --> C(Implemented Model)
  B <-. Verification .-> C
  A <-. Validation .-> C
```

```mermaid
flowchart LR
  A(Real World) -- Design --> B(Conceptual Model)
  B -- Construction --> C(Implemented Model)
  B <-. Verification .-> C
  A <-. Validation .-> C
```

**Comment by stereobooster** [comment]
maybe [graphviz](https://dreampuf.github.io/GraphvizOnline/#digraph%20G%20%7B%0A%20%20rankdir%20%3D%20TD%3B%0A%0A%20%20A%5Blabel%3D%22Real%20World%22%20shape%3Dbox%5D%0A%20%20B%5Blabel%3D%22Conceptual%20Model%22%20shape%3Dbox%5D%0A%20%20C%5Blabel%3D%22Implemented%20Model%22%20shape%3Dbox%5D%0A%20%20%0A%20%20A%20-%3E%20B%20%5Blabel%3D%22Design%22%5D%3B%0A%20%20B%20-%3E%20C%20%5Blabel%3D%22Construction%22%5D%3B%0A%20%20C%20-%3E%20B%20%5Blabel%3D%22Verification%22%5D%3B%0A%20%20C%20-%3E%20A%20%5Blabel%3D%22Validation%22%5D%3B%0A%20%0A%20%20%7B%20rank%20%3D%20same%3B%20A%20B%20C%20%7D%0A%7D)

**Reply by danielvartan** [reply]
Very nice solution. Thank you!

Although it's not perfect, there must be a way to do this.

---

```dot
digraph G {
  rankdir = TD;

  A[label="Real World" shape=box]
  B[label="Conceptual Model" shape=box]
  C[label="Implemented Model" shape=box]
  
  A -> B [label="Design"];
  B -> C [label="Construction"];
  C -> B [label="Verification"];
  C -> A [label="Validation"];
 
  { rank = same; A B C }
}
```

![image](https://github.com/user-attachments/assets/59843d88-84c4-4ca0-893d-75ac9c8ef346)
